City-of-Helsinki/hel-fi-drupal-grants
# feat: AU-2049: E2E, Application print content test

# [AU-2049](https://helsinkisolutionoffice.atlassian.net/browse/AU-2049)

## What was done

This pull request implements an E2E test for validating the content on an applications "print" page (`/fi/hakemus/{APPLICATION_ID}/tulosta`). The test makes sure that the submitted application content is present on the print page. The print test has been added to the `copy` form variant test on the forms `64` and `67`.

## What has changed

- A new `Validate print page` test has been added to each test file. The test creates a new browser context where JavaScript has been disabled. This is because Playwright isn't able to handle the print dialog, so we have to disable JS in order to disable the printing functionality that's automatically triggered when visiting the `/tulosta` page. After this, the test simply validates the the submitted application content is present on the print page. The validation is done in the same manner as the validation on the `/katso` page.

-  A new `validatePrintPage` boolean key has been added to the `FormData` interface. Setting the value to `true` on a form variant makes the `Validate print page` test fire for said variant. 

-  A new `printPageSkipValidation` boolean key has been added to the `FormField` interface. Setting the value to `true` on a form field makes the validation test skip said field when validating the content on the print page.

## How to install

Make sure your instance is up and running on correct branch.
* `feature/AU-2049-e2e-application-print`
* `make fresh`
* `make drush-cr`

## How to test

- Remove `ENABLED_FORM_VARIANTS` and `DISABLED_FORM_VARIANTS` from the `.env` file if you want to run all form variants. If you're in a hurry, you can set `ENABLED_FORM_VARIANTS="copy"`.

- Run `make test-pw-p PROJECT=forms-64 forms-67`. Make sure all tests passes and that you see this during the `Validate print page` test:

<img width="1568" alt="Screenshot 2024-05-29 at 15 49 04" src="https://github.com/City-of-Helsinki/hel-fi-drupal-grants/assets/117807015/4255ebc6-6ff6-4b3f-88dc-fe56825c5dd9">

<img width="1406" alt="Screenshot 2024-05-29 at 15 50 34" src="https://github.com/City-of-Helsinki/hel-fi-drupal-grants/assets/117807015/93054017-253e-429d-b807-d63af497c21c">





[AU-2049]: https://helsinkisolutionoffice.atlassian.net/browse/AU-2049?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
